android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountSelectorActivity.java

```diff
@@ -92,4 +70,40 @@ public void onAccountClick(WalletListItemModel walletListItemModel) {
             }
         }
     }
+
+    @SuppressLint("NotifyDataSetChanged")
+    private void initAccounts() {
+        mKeyringModel.getAccounts(accountInfos -> {
+            mAccountInfos = accountInfos;
+            List<WalletListItemModel> walletListItemModelList = new ArrayList<>();
+            for (AccountInfo accountInfo : mAccountInfos) {
+                if (!accountInfo.isImported) {
+                    walletListItemModelList.add(
+                            new WalletListItemModel(R.drawable.ic_eth, accountInfo.name,
+                                    accountInfo.address, null, null, accountInfo.isImported));
+                }
+            }
+
+            mWalletCoinAdapter.setWalletListItemModelList(walletListItemModelList);
+            mWalletCoinAdapter.notifyDataSetChanged();
+
+            mKeyringModel.mSelectedAccount.observe(this, selectedAccountInfo -> {
+                if (selectedAccountInfo == null) return;
+
+                mWalletCoinAdapter.updateSelectedNetwork(
+                        selectedAccountInfo.name, selectedAccountInfo.address);
+                boolean callDataSetChanged = true;
+                for (AccountInfo accountInfo : mAccountInfos) {
+                    if (selectedAccountInfo.name.equals(accountInfo.name)
```

Comment:
Can two accounts have the same address but different names? Because https://github.com/brave/brave-core/blob/aca7b23d821eb65152ec314e8962bdf3858ea34e/android/java/org/chromium/chrome/browser/app/domain/KeyringModel.java#L93 only check addresses but here it checks both names and addresses, one of them need to be changed I think.

Reply:
I just tried and we don't let to add any new account with same address, changing to check on address only.

Reply:
I updated it

Reply:
Ok lgtm.